extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -423,36 +423,57 @@ static public function setmainDomain ( $city_id, $domain = null, $reason = null
 	 * @access public
 	 * @static
 	 *
-	 * @param $url string - domain name
+	 * @param $url string - domain name in form:
+	 * - http://www.community-name.wikia.com
+	 * - http://community-name.wikia.com
+	 * - www.community-name.wikia.com
+	 * - community-name.wikia.com
+	 * - community-name
 	 *
 	 * @return integer - id of domain or null if not found
 	 */
 	static public function UrlToID( $url ) {
-
 		$city_id = false;
+
+		$url = self::prepareUrlToParse( $url );
 		$parts = parse_url( $url );
+
 		if ( isset( $parts[ "host" ] ) ) {
 			$host = self::getDomainHash( $parts[ "host" ] );
-
-			// TODO: Eloy: Can this hack be removed?
-			if ( $host === "memory-alpha.org" ) {
-				/**
-				 * for memory-alpha check first element of path
-				 */
-				$parts = explode( "/", $parts[ "path" ] );
-				$host = sprintf( "%s.%s", $parts[ 1 ], $host );
-				$city_id = self::DomainToId( $host );
-			}
-			else {
-				$host = preg_replace('/^(?:preview\.|verify\.)/i', '', $host);
-				$city_id = self::DomainToId( $host );
-			}
+			$host = preg_replace('/^(?:preview\.|verify\.)/i', '', $host);
+			$city_id = self::DomainToId( $host );
 		}
 
 		return $city_id;
 	}
 
 	/**
+	 * Convert url in form of:
+	 * - www.community-name.wikia.com
+	 * - community-name.wikia.com
+	 * - community-name
+	 * to full valid url.
+	 * See: testPrepareUrlToParse
+	 *
+	 * @param $url
+	 * @return string
+	 */
+	static public function prepareUrlToParse( $url ) {
+		$httpPrefix = 'http://';
+		$wikiacomSuffix = '.wikia.com';
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal ".wikia.com" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)